src/lua/log.lua

```diff
@@ -636,6 +645,123 @@ local function load_cfg(self, cfg)
     say(S_DEBUG, m:format(cfg.log, cfg.level, cfg.nonblock, cfg.format))
 end
 
+local option_types = {
+    log = 'string',
+    nonblock = 'boolean',
+    level = 'number, string',
+    format = 'string',
+}
+
+local log_initialized = false
+
+local function log_C_cfg(cfg)
```

Comment:
Nit: It'd be nice to have brief comments for this function and all the new helper functions below. They aren't quite trivial so comments would help.

Reply:
Done.